src/interpolations.jl

```diff
@@ -457,6 +490,75 @@ is_discontinuous(::Type{<:DiscontinuousLagrange}) = true
+# ...
+
+# Sum factorized eval for tensor product elements
+# function shape_values!(values::AT, ip::Lagrange{RefHypercube{2},order}, ξ::Vec{2,T}) where {T, AT <: AbstractArray, order}
+#     # Auxillary 1D interpolation
+#     ip_1d = Lagrange{RefLine,order}()
+#     n_basefunctions_1d = getnbasefunctions(ip_1d)
+#     # Evaluate 1D
+#     evals_x_1d = ntuple(i->shape_value(ip_1d, Vec{1,T}((ξ[1],)), i), order+1)
+#     evals_y_1d = ntuple(i->shape_value(ip_1d, Vec{1,T}((ξ[2],)), i), order+1)
+#     # Sum factorization
+#     values_2d = reshape(values, (n_basefunctions_1d, n_basefunctions_1d))
+#     for ix in 1:n_basefunctions_1d, iy in 1:n_basefunctions_1d
+#         values_2d[ix, iy] = evals_x_1d[ix] * evals_y_1d[iy]
+#     end
+#     TODO permutate values_2d
+# end
+
+# function shape_values!(gradients_2d::AT, ip::Lagrange{RefHypercube{2},order}, ξ::Vec{2,T}) where {T, AT <: AbstractArray, order}
+#     # Auxillary 1D interpolation
+#     ip_1d = Lagrange{RefLine,order}()
+#     n_basefunctions_1d = getnbasefunctions(ip_1d)
+#     # Evaluate 1D
+#     evals_x_1d = ntuple(i->shape_value(ip_1d, Vec{1,T}((ξ[1],)), i), order+1)
+#     evals_y_1d = ntuple(i->shape_value(ip_1d, Vec{1,T}((ξ[2],)), i), order+1)
+#     evals_dx_1d = ntuple(i->shape_gradient(ip_1d, Vec{1,T}((ξ[1],)), i), order+1)
+#     evals_dy_1d = ntuple(i->shape_gradient(ip_1d, Vec{1,T}((ξ[2],)), i), order+1)
+#     # Sum factorization
+#     gradients_2d_2d = reshape(gradients_2d, (n_basefunctions_1d, n_basefunctions_1d))
+#     for ix in 1:n_basefunctions_1d, iy in 1:n_basefunctions_1d
+#         gradients_2d_2d[ix, iy, 1] = ...
+#         gradients_2d_2d[ix, iy, 2] = ...
+#     end
+#     TODO permutate gradients_2d_2d
+# end
+
+# function shape_gradients_and_values!(gradients::GAT, shapes::SAT, ip::Lagrange{RefHypercube{2},order}, ξ::Vec ) where {T, SAT <: AbstractArray, GAT <: AbstractArray, dim, order}
+#     # Auxillary 1D interpolation
+#     ip_1d = Lagrange{RefLine,order}()
+#     n_basefunctions_1d = getnbasefunctions(ip_1d)
+#     # Evaluate 1D
+#     evals_x_1d = ntuple(i->shape_value(ip_1d, Vec{1,T}((ξ[1],)), i), order+1)
+#     evals_y_1d = ntuple(i->shape_value(ip_1d, Vec{1,T}((ξ[2],)), i), order+1)
+#     evals_dx_1d = ntuple(i->shape_gradient(ip_1d, Vec{1,T}((ξ[1],)), i), order+1)
+#     evals_dy_1d = ntuple(i->shape_gradient(ip_1d, Vec{1,T}((ξ[2],)), i), order+1)
+#     # Sum factorization
+#     values_2d    = reshape(values, (n_basefunctions_1d, n_basefunctions_1d))
+#     gradients_2d = reshape(gradients, (n_basefunctions_1d, n_basefunctions_1d))
+#     for ix in 1:n_basefunctions_1d, iy in 1:n_basefunctions_1d
+#         values_2d[ix, iy, 1] = ...
+#         values_2d[ix, iy, 2] = ...
+#     end
+#     TODO permutate values_2d
+#     TODO permutate gradients_2d_2d
+# end
+
+# TODO implement kernels above for 3D
+# TODO implement vectorized interpolation case
+
```

Comment:
Should this be removed before merge?

Reply:
I think so. This is basically a template to guide how the evals might be implemented in https://github.com/Ferrite-FEM/Ferrite.jl/pull/790 .